src/stash/src/upgrade.rs

```diff
@@ -73,3 +126,46 @@ where
         Ok(())
     }
 }
+
+impl TypedCollection<ConfigKey, ConfigValue> {
+    pub async fn version(&self, tx: &mut Transaction<'_>) -> Result<u64, StashError> {
+        let key = ConfigKey {
+            key: USER_VERSION_KEY.to_string(),
+        };
+        let config = COLLECTION_CONFIG.from_tx(tx).await?;
+        let version = tx
+            .peek_key_one(config, &key)
+            .await?
+            .ok_or_else(|| StashError {
+                inner: InternalStashError::Uninitialized,
+            })?;
+
+        Ok(version.value)
+    }
+
+    pub async fn set_version(
+        &self,
+        tx: &mut Transaction<'_>,
+        version: u64,
+    ) -> Result<(), StashError> {
+        let key = ConfigKey {
+            key: USER_VERSION_KEY.to_string(),
+        };
+        let value = ConfigValue { value: version };
+
+        // Either insert a new version, or bump the old version.
+        COLLECTION_CONFIG
+            .migrate_to(tx, |entries| {
+                let action = if entries.contains_key(&key) {
+                    MigrationAction::Update(key.clone(), (key, value))
+                } else {
+                    MigrationAction::Insert(key, value)
```

Comment:
Another interesting case that could probably be tested by not having a previous version and thus inserting a new one instead of updating.